sandbox/graduation/graduation_scratch.R

```diff
@@ -0,0 +1,695 @@
+#'                                                                        total_remedial_hours,
+#'                                                                        level_id,
+#'                                                                        previous_degree_type,
+#'                                                                        ipeds_award_level_code,
+#'                                                                        required_credits,
+#'                                                                        high_school_code,
+#'                                                                        ssid,
+#'                                                                        earned_contact_hrs ,
+#'                                                                        program_hrs,
+#'                                                                        graduated_term_id,
+#'                                                                        financial_aid_year_id,
+#'                                                                        season,
+#'                                                                        primary_major_college_id,
+#'                                                                        primary_major_desc,
+#'                                                                        degree_desc)
+#'
+#'
+#'
+#' @param with_intermediates Boolean: Option to include intermediate calculated fields.
+#'
+#' @return
+#' A Data Frame, with all of the USHE elements required for upload submission.
+#' This will also include intermediate values, used to calculate USHE data elements, if option is set.
+#'
+#' @examples
+#' generate_graduation_submission_file()
+#'
+#' @export
+#'
+generate_graduation_submission_file <- function(input_df=usheUtils::fake_graduation_df, with_intermediates=FALSE) {
+
+  ushe_data_elements <- c("g_01", "g_02", "g_03", "g_04",
+                          "g_05", "g_06", "g_07", "g_08",
+                          "g_09", "g_10", "g_11", "g_12",
+                          "g_13", "g_14", "g_15", "g_16",
+                          "g_17", "g_18", "g_19", "g_20",
+                          "g_21", "g_22", "g_23", "g_24",
+                          "g_25", "g_26", "g_27", "g_28")
+
+  output_df <- input_df %>%
+    g_01() %>%
+    g_02() %>%
+    g_03() %>%
+    g_04() %>%
+    g_05() %>%
+    g_06() %>%
+    g_07() %>%
+    g_08() %>%
+    g_09() %>%
+    g_10() %>%
+    g_11() %>%
+    g_12() %>%
+    g_13() %>%
+    g_14() %>%
+    g_15() %>%
+    g_16() %>%
+    g_17() %>%
+    g_18() %>%
+    g_19() %>%
+    g_20() %>%
+    g_21() %>%
+    g_22() %>%
+    g_23() %>%
+    g_24() %>%
+    g_25() %>%
+    g_26() %>%
+    g_27() %>%
+    g_28()
+
+
+  if (!with_intermediates) {
+    output_df <- output_df %>%
+      dplyr::select( ushe_data_elements )
+  }
+
+  return(output_df)
+}
+
```

Comment:
@cdemkegit Where is g_01 through g_07? Will these now be general based functions?